packages/phpstan-rules/tests/Rules/ForbiddenFuncCallRule/config/configured_rule_with_deprecations.neon

```diff
@@ -0,0 +1,13 @@
+includes:
+    - ../../../../config/services/services.neon
+
+services:
+    -
+        class: Symplify\PHPStanRules\Rules\ForbiddenFuncCallRule
+        tags: [phpstan.rules.rule]
+        arguments:
+            forbiddenFunctions:
+                'extract': 'you shouldn"t use this dynamic things'
+                'dump': 'seems you missed some debugging function'
+                'curl_*': 'we use a proper http client these days'
+                'property_exists': ''
```

Comment:
Just curious, what happens if user writes syntax? 

```suggestion
                - 'property_exists'
```

It's quite common to write NEON/YAML lists this way

Reply:
I had implemented it that way initially. it changes the structure of the forbiddenFunctions parameter to be a list of arrays instead of just a simple map.

it will not work as it was implemented right now.

I changed to the simpler form, because its the way all other config key I found also work, so we are consistent at this end

Reply:
I'm not sure I understand. In the end both forms have to work, because people will try them both :)

Reply:
sementically 

```
forbiddenFunctions:
- 'extract': 'you shouldn"t use this dynamic things'
- 'dump': 'seems you missed some debugging function'
```

and

```
forbiddenFunctions:
  'extract': 'you shouldn"t use this dynamic things'
  'dump': 'seems you missed some debugging function'
```

are totally different things. 


as another counter example - in phpstan the following 2 things are not equivalent and don't work both
```
services:
    -
        class: Symplify\PHPStanRules\Rules\ForbiddenFuncCallRule
        tags: [phpstan.rules.rule]
```    

```
services:
    - class: Symplify\PHPStanRules\Rules\ForbiddenFuncCallRule
    - tags: [phpstan.rules.rule]
```

Reply:
Exactly, they both need to be supported. It's valid NEON syntax, see:

https://ne-on.org/#saved-K0ktLrHiUgCC7NRKK4WyxJzSVDBXVyExL78kI7UoHigBAA==